pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PqlQueryResource.java

```diff
@@ -76,8 +76,12 @@ public String post(String requestJsonStr, @Context HttpHeaders httpHeaders) {
       if (requestJson.has("trace")) {
         traceEnabled = requestJson.get("trace").toString();
       }
+      String queryOptions = null;
+      if (requestJson.has("queryOptions")) {
+        queryOptions = requestJson.get("queryOptions").asText();
+      }
       LOGGER.debug("Trace: {}, Running query: {}", traceEnabled, pqlQuery);
-      return getQueryResponse(pqlQuery, traceEnabled, httpHeaders);
+      return getQueryResponse(pqlQuery, traceEnabled, queryOptions, httpHeaders);
```

Comment:
traceEnabled can also be part of queryOptions rt

Reply:
If we move trace to queryOptions, the callers of the broker API will have to be changed to pass trace in queryOptions.